hcn/hcnnamespace.go

```diff
@@ -63,6 +64,7 @@ type HostComputeNamespace struct {
 	Type          NamespaceType       `json:",omitempty"` // Host, HostDefault, Guest, GuestDefault
 	Resources     []NamespaceResource `json:",omitempty"`
 	SchemaVersion SchemaVersion       `json:",omitempty"`
+	ReadyOnCreate bool                `json:",omitempty"`
```

Comment:
What does `ReadyOnCreate` do?

Reply:
new HNS code that does compartment initialization on pod create needs this new flag to kick in the new code path

Reply:
https://microsoft.visualstudio.com/OS/_git/os.2020/pullrequest/9670361